src/morphology.rs

```diff
@@ -511,13 +514,12 @@ impl Mask {
     /// # }
     /// ```
     pub fn diamond(radius: u8) -> Self {
-        Self {
-            elements: (-(radius as i16)..=(radius as i16))
-                .flat_map(|x| {
-                    ((x.abs() - radius as i16)..=(radius as i16 - x.abs())).map(move |y| (x, y))
-                })
-                .collect(),
-        }
+        let mut elements = Vec::new();
+        let mut iter = (-(radius as i16)..=(radius as i16)).flat_map(|y| {
+            ((y.abs() - radius as i16)..=(radius as i16 - y.abs())).map(move |x| Point::new(x, y))
+        });
+        elements.resize_with(1 + 2 * (radius as usize) * (radius as usize + 1), || iter.next().expect("the size of iter is exactly equal to 1 + 2 * radius * (radius + 1) (i, Morgane55440, did the math)"));
```

Comment:
Don't panic. You can try `Vec::with_capacity` and then for loop with `push`.

Reply:
i have done something similar, but with `extend` instead of `push`